test/expect-tests/dune_pkg/dune_pkg_better_candidate_test.ml

```diff
@@ -0,0 +1,360 @@
+open Stdune
+open Dune_pkg
+open Dune_lang
+module Console = Dune_console
+
+let create_dummy_package name version : Lock_dir.Pkg.t =
+  { Lock_dir.Pkg.build_command = None
+  ; install_command = None
+  ; exported_env = []
+  ; deps = []
+  ; info =
+      { Lock_dir.Pkg_info.name = Package_name.of_string name
+      ; version
+      ; dev = false
+      ; source = None
+      ; extra_sources = []
+      }
+  }
+;;
+
+let dummy_results
+  number_of_immediate
+  total_number_of_immediate
+  number_of_transitive
+  total_number_of_transitive
+  =
+  List.init (total_number_of_immediate - number_of_immediate) ~f:(fun _ ->
+    { Better_candidate.is_immediate_dep_of_local_package = true
+    ; result = Package_is_best_candidate
+    ; pkg = create_dummy_package "foo" "1.0.0"
+    })
+  @ List.init number_of_immediate ~f:(fun _ ->
+    { Better_candidate.is_immediate_dep_of_local_package = true
+    ; result = Better_candidate { version = OpamPackage.Version.of_string "2.0.0" }
+    ; pkg = create_dummy_package "foo" "1.0.0"
+    })
+  @ List.init (total_number_of_transitive - number_of_transitive) ~f:(fun _ ->
+    { Better_candidate.is_immediate_dep_of_local_package = false
+    ; result = Package_is_best_candidate
+    ; pkg = create_dummy_package "bar" "2.0.0"
+    })
+  @ List.init number_of_transitive ~f:(fun _ ->
+    { Better_candidate.is_immediate_dep_of_local_package = false
+    ; result = Better_candidate { version = OpamPackage.Version.of_string "3.0.0" }
+    ; pkg = create_dummy_package "bar" "2.0.0"
+    })
+;;
+
+let find_color_of_line line =
```

Comment:
How about you just strip the colors from the output in these tests? It doesn't seem like it's worth testing.

Reply:
I included them so that we can see exactly what colours the messages are, as there are a lot of them.

Reply:
Its also the kind of thing that we will never see during development but users will see all the time. I therefore think its important to have some idea of what users will end up seeing.